src/main/java/edu/hillel/homework/lesson17/petrol_station/PetrolStation.java

```diff
@@ -0,0 +1,84 @@
+package edu.hillel.homework.lesson17.petrol_station;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import edu.hillel.homework.lesson17.petrol_station.car.Car;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+public class PetrolStation {
+
+    private static final int MAXIMUM_THREADS_ALLOWED = 3;
+    private final AtomicDouble amount;
+    private final ExecutorService executor;
+
+    public PetrolStation(double initialFuelAmount) {
+        this.amount = new AtomicDouble(initialFuelAmount);
+        ThreadFactory threadFactory = new ThreadFactory() {
+            private int pumpNumber = 1;
+
+            @Override
+            public @NotNull Thread newThread(@NotNull Runnable r) {
+                return new Thread(r, Integer.toString(pumpNumber++));
+            }
+        };
+        this.executor = Executors.newFixedThreadPool(MAXIMUM_THREADS_ALLOWED, threadFactory);
+    }
+
+    public double getAmount() {
+        return amount.get();
+    }
+
+    public void setAmount(double amount) {
+        this.amount.addAndGet(amount);
+    }
+
+    public void doRefuel(Car car, double fuelAmount) {
+        executor.execute(() -> {
+            double newAmount = amount.get() - fuelAmount;
```

Comment:
You can try
amount.updateAndGet(amount -> amount - fuelAmount);
And it will be wrapped operation by atomic function